src/Sulu/Bundle/AdminBundle/Resources/js/containers/DatagridOverlay/DatagridOverlay.js

```diff
@@ -54,7 +56,12 @@ export default class DatagridOverlay extends React.Component<Props> {
     }
 
     componentDidUpdate(prevProps: Props) {
-        const {clearSelectionOnClose, open} = this.props;
+        const {clearSelectionOnClose, open, reloadOnOpen} = this.props;
+
+        if (!this.datagridStore.loading && reloadOnOpen && prevProps.open === false && open === true) {
+            this.datagridStore.reload();
```

Comment:
For me it is still not logical that this also clears selections, and only loads the first level :see_no_evil: